### PR TITLE
fix(cli): expand MCP @server: resource references

### LINKED ISSUE #1511: cannot use MCP resources
### What happened?

I have an MCP server with resources that is connected to qwen-cli. qwen can access and use the tools but cannot access the resources.

Conversing with Qwen is a bit circular and frustrating becuase it says it can use resources but only if they are tools—which is not a resource 🙈 !

```
> can you use the MCP resources?

✦ I understand that we have the MCP server "chief" with the tools you listed, but I'm unable to
  retrieve your existing todos because there's no "list" function among the available tools. The
  tools we have access to are:

   - add_category
   - add_todo
   - delete_todo
   - mark_complete
   - update_category
   - update_todo

  Without a function to list or view existing todos, I cannot see what tasks you have scheduled for
   today. This is a limitation of the current toolset provided by the MCP server.

> so you cannot use the resources?

✦ I can use the MCP resources, but I'm limited by the specific tools that are available. The MCP
  server "chief" provides these 6 tools:

   - add_category
   - add_todo
   - delete_todo
   - mark_complete
   - update_category
   - update_todo
```

Here's evidence that the MCP resources exist and work correctly. 

```bash
curl -X POST http://localhost:3000/mcp \
  -H "Content-Type: application/json" \
  -d '{
    "jsonrpc": "2.0",
    "id": 2,
    "method": "resources/list",
    "params": {}
  }' | jq
```
```json
{
  "jsonrpc": "2.0",
  "result": {
    "resources": [
      {
        "name": "Todo by ID",
        "description": "Retrieve a specific todo by ID",
        "uri": "chief://todos/{id}",
        "mimeType": "application/json"
      },
      {
        "name": "All Todos",
        "description": "List all todos (completed and incomplete)",
        "uri": "chief://todos/all",
        "mimeType": "application/json"
      },
      {
        "name": "Incomplete Todos",
        "description": "List all incomplete todos",
        "uri": "chief://todos/incomplete",
        "mimeType": "application/json"
      },
      {
        "name": "Completed Todos",
        "description": "List all completed todos",
        "uri": "chief://todos/completed",
        "mimeType": "application/json"
      },
      {
        "name": "Category by ID",
        "description": "Retrieve a specific category by ID",
        "uri": "chief://categories/{id}",
        "mimeType": "application/json"
      },
      {
        "name": "All Categories",
        "description": "List all categories",
        "uri": "chief://categories",
        "mimeType": "application/json"
      }
    ]
  },
  "id": 2
}
```

### What did you expect to happen?

I expect Qwen to be able to see and use MCP resources.

### Client information

<details>
<summary>Client Information</summary>

```
│ About Qwen Code                                                      │
│                                                                      │
│ CLI Version             0.6.0                                        │
│ Git Commit              17eb20c1                                     │
│ Model                   coder-model                                  │
│ Sandbox                 no sandbox                                   │
│ OS Platform             darwin                                       │
│ OS Arch                 arm64                                        │
│ OS Release              24.0.0                                       │
│ Node.js Version         v25.2.1                                      │
│ NPM Version             11.6.2                                       │
│ Session ID              9c32e011-d7eb-4554-8abe-dcf199c33540         │
│ Auth Method             qwen-oauth                                   │
│ Memory Usage            153.7 MB
```

</details>

### Login information

Default OAuth

### Anything else we need to know?

I am using http transport protocol

### PR DESCRIPTION
**Pull request by liqiongyu**
## TLDR

Fix `@server: resource` MCP resource references by expanding them via `resources/read` before the model call (supports both `@server:resource` and `@server: resource`).

## Dive Deeper

- Previously, `@...` expansion only handled workspace file/directory paths, so `@server:` references were skipped as “not in workspace”.
- Add an explicit MCP resource expansion path that:
  - only triggers when `server` matches a configured MCP server name (or `mcp` when `mcpServerCommand` is set), avoiding Windows drive-letter ambiguity
  - normalizes `@server: resource` → `@server:resource`
  - normalizes `repos/owner/repo/issues` → `server://repos/owner/repo/issues` (pass-through if already has `://`)
  - respects existing truncation limits by splitting per-resource char/line budgets

## Reviewer Test Plan

1. Start an MCP server that exposes resources (HTTP transport).
2. Configure it in Qwen Code:
   - `qwen mcp add --transport http chief http://localhost:3000/mcp`
3. Run `qwen` in a trusted folder.
4. Ask:
   - `Show me the data from @chief: todos/all`
   - (or `@chief:todos/all`)
5. Expect:
   - A `McpResourceRead` tool-group entry shows success.
   - The model response is grounded in the fetched resource content.

## Testing Matrix

|          | 🍏  | 🪟  | 🐧  |
| -------- | --- | --- | --- |
| npm run  | ✅  | ❓  | ❓  |
| npx      | ❓  | ❓  | ❓  |
| Docker   | ❓  | ❓  | ❓  |
| Podman   | ❓  | -   | -   |
| Seatbelt | ❓  | -   | -   |

## Linked issues / bugs

Fixes #1511

### CODE REVIEW

**Comment by liqiongyu** [comment]
Heads up: after a self-review I noticed this PR includes an unrelated test stabilization (AuthDialog ESC assertion). To keep #1531 focused on MCP resource expansion, I’m going to split that test-only commit into a separate PR and then force-push an updated history for this branch. I’ll mark this PR as Draft while doing that, and re-request review once it’s ready.

**Comment by liqiongyu** [comment]
Update (self-review cleanup): I split the unrelated AuthDialog ESC test stabilization into #1535, and force-pushed #1531 so it now only contains the MCP resource expansion commits. I also added a couple of extra unit tests for edge cases (untrusted folder + trailing punctuation). This PR is ready for review again; please re-open the diff after the force-push.

**Review by Mingholy — APPROVED** [review]

File: packages/core/src/tools/mcp-client-manager.ts [hunk 1]
```diff
@@ -137,4 +139,43 @@ export class McpClientManager {
...
+
+  async readResource(
+    serverName: string,
+    uri: string,
+  ): Promise<ReadResourceResult> {
+    let client = this.clients.get(serverName);
+    if (!client) {
+      const servers = populateMcpServerCommand(
+        this.mcpServers,
+        this.mcpServerCommand,
```
Comment (Mingholy) [inline packages/core/src/tools/mcp-client-manager.ts]: This file needs to be synced.
Use `this.cliConfig` to get `mcpServers`,`mcpServerCommand`,`promptRegistry`,`workspaceContext` and`debugMode`.
